substrate/client/network/src/discovery.rs

```diff
@@ -785,16 +803,24 @@ impl NetworkBehaviour for DiscoveryBehaviour {
 							Ok(GetRecordOk::FoundRecord(r)) => {
 								debug!(
 									target: "sub-libp2p",
-									"Libp2p => Found record ({:?}) with value: {:?}",
+									"Libp2p => Found record ({:?}) with value: {:?} id {:?} stats {:?}",
 									r.record.key,
 									r.record.value,
+									id,
+									stats,
 								);
 
-								// Let's directly finish the query, as we are only interested in a
-								// quorum of 1.
-								if let Some(kad) = self.kademlia.as_mut() {
-									if let Some(mut query) = kad.query_mut(&id) {
-										query.finish();
+								// Let's directly finish the query if we are above 4.
+								// This number should small enough to make sure we don't
+								// unnecessarily flood the network with queries, but high
+								// enough to make sure we also touch peers which might have
+								// old record, so that we can update them once we notice
+								// they have old records.
+								if stats.num_successes() > 4 {
```

Comment:
Would be nice to make this configurable per `get_record`/`get_value` request, and only pass > 1 for authority-discovery at the caller side. At least something to consider as a TODO for a follow-up PR. In this PR it's worth extracting 4 as something like `GET_RECORD_REDUNDANCY_FACTOR` to the beginning of `discovery.rs`.

Reply:
Will do.